File: 1103-moving-stones-until-consecutive/moving-stones-until-consecutive.java
class Solution {
    public int[] numMovesStones(int a, int b, int c) {
        int max = Math.max(c,Math.max(a,b)), min = Math.min(c,Math.min(a,b));
        int max1 = max - min - 2;
        int min1 = Math.max(Math.min(3, Math.min(Math.abs(a - b), Math.min(Math.abs(b - c), Math.abs(a - c)))) - 1, 1);
        min1 = (max1 == 0) ? 0 : min1;
        return new int[] {min1, max1};
    }
}

File: 1103-moving-stones-until-consecutive/moving-stones-until-consecutive.rs
impl Solution {
    pub fn num_moves_stones(a: i32, b: i32, c: i32) -> Vec<i32> {
        let max_val = std::cmp::max(c, std::cmp::max(a, b));
        let min_val = std::cmp::min(c, std::cmp::min(a, b));
        let max1 = max_val - min_val - 2;
        let min1 = std::cmp::max(
            std::cmp::min(
                3,
                std::cmp::min(
                    (a - b).abs(),
                    std::cmp::min((b - c).abs(), (a - c).abs()),
                ),
            ) - 1,
            1,
        );
        let min1 = if max1 == 0 { 0 } else { min1 };
        vec![min1, max1]
    }
}